docs/getting-started/docker-compose.yaml

```diff
@@ -4,6 +4,11 @@ services:
     image: ghcr.io/sovity/edc-ui:latest
     ports:
       - '11000:80'
+    healthcheck:
+      test: [ "CMD", "curl", "-f", "http://localhost:80/" ]
+      interval: 2s
+      timeout: 5s
+      retries: 10
```

Comment:
In Docker you can define a healthcheck in the Dockerfile with a HEALTHCHECK directive.

It is best practice to supply it and we do.

However, docker-compose files do not respect that original HEALTHCHECK directive, which causes our health checks to be untested.

The EDC UI health check was reported broken by people who deployed it with docker swarm. They needed to manually disable it in the image in the end.

Thus we should double check the HEALTHCHECK statement in the Dockerfile. To actually test the Dockerfile's HEALTHCHECK directive, we might require either docker swarm or kubernetes to launch the container in a way, that its original healtcheck actually gets checked.

Reply:
> To actually test the Dockerfile's HEALTHCHECK directive, we might require either docker swarm or kubernetes to launch the container in a way, that its original healtcheck actually gets checked.

Unfortunately, I was unable to find conclusive documentation on this, but as far as I am aware, there is no way to make Kubernetes honor the `HEALTHCHECK` instruction.
Health checks are a Docker-specific feature; the underlying runtime, containerd, does not support them natively. Kubernetes does not use Docker, instead accessing containerd (or comparable runtimes) directly.

Reply:
That's an option, yes.